src/game/client/water/w3dwatertracks.cpp

```diff
@@ -325,9 +329,204 @@ int WaterTracksObj::Update(int ms_elapsed)
 
 void Test_Water_Update()
 {
-    // todo needs InGameUI
-#ifdef GAME_DLL
-    Call_Function<void>(PICK_ADDRESS(0x007A22C0, 0x0064DC2D));
+#ifdef BUILD_WITH_D3D8
+    static bool pause_waves;
+    static POINT mouse_anchor;
+    static Coord3D terrain_point_start;
+    static Coord3D terrain_point_end;
+    static WaterTracksObj *track;
+    static WaterTracksObj *track2;
+    static int track_edit_mode;
+    static int have_start;
+    static int have_end;

```

Comment:
Wow. That is a lot of statics. I think it would be best to give them all default values.

Reply:
fixed.